File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        unordered_map<int,int>mp;

        for(int i=0;i<n;++i){
            int compliment=target-nums[i];
            if(mp.find(compliment)==mp.end()){
                mp[nums[i]]=i;
            }
            else return {mp[compliment],i};
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n=nums.length;
        HashMap<Integer,Integer> mp=new HashMap<>();

        for(int i=0;i<n;++i){
            if(!mp.containsKey(target-nums[i])){
                mp.put(nums[i],i);
            }
            else{
                return new int[]{mp.get(target-nums[i]),i};
            }
        }
        return new int[] {};
    }
}